tests/spread/testcraft/fetch-service-external/task.yaml

```diff
@@ -0,0 +1,97 @@
+summary: Testcraft can use an external fetch service session
+
+priority: 100
+
+prepare: |
+  snap install --candidate fetch-service
+
+  # Some of these values match the current defaults used by the fetch-service, but we
+  # set them here too to protect against changes in those defaults
+  snap set fetch-service control.auth="craft:craft"
+  snap set fetch-service control.port=9999
+  snap set fetch-service proxy.port=9988
+  snap set fetch-service permissive=true
+  snap set fetch-service log.file="from-testcraft.log"
+  snap start fetch-service
+
+  # Give the service some time to start
+  sleep 5
+  # Make sure the service is up
+  curl -f http://localhost:9999/status
+
+  # Create a fetch-service session
+  curl -f -X POST -d '{"policy": "strict"}' http://craft:craft@localhost:9999/session --output session.json
+  session_id=$(jq -r .id session.json)
+  token=$(jq -r .token session.json)
+
+  # This assumes that the lxd network is lxdbr0
+  host_ip=$(ip -f inet addr show lxdbr0 | sed -En -e 's/.*inet ([0-9.]+).*/\1/p')
+
+  # Save the proxy url so that the 'execute' step can access it.
+  echo "http://${session_id}:${token}@${host_ip}:9988" > session_proxy_url.txt
+
+execute: |
+  # Configure testcraft to use the external session
+  export CRAFT_USE_EXTERNAL_FETCH_SERVICE=1
+
+  # Check that the fetch-service's certificate is there, use it
+  test -f /var/snap/fetch-service/current/local-ca.crt
+  export CRAFT_PROXY_CERT=/var/snap/fetch-service/current/local-ca.crt
+
+  # Set http_proxy and https_proxy variables to the value saved during 'prepare'
+  export http_proxy=$(cat session_proxy_url.txt)
+  export https_proxy=$(cat session_proxy_url.txt)
+
+  # The current implementation is bugged: when setting up the lxd instance snapd is
+  # configured to use the proxy but it doesn't (can't) recognize the fetch-service
+  # signature because we don't inject the certificate.
+  expected_error='cannot install system snap "snapd": Post "https://api.snapcraft.io/v2/snaps/refresh": tls: failed to verify certificate: x509: certificate signed by unknown authority'
```

Comment:
this right here is the current (but not the only) problem: craft-providers is configuring snapd to use the proxy because `http_proxy` is set, but not the fetch-service certificate is unavailable at that time